src/gsy_e/models/strategy/predefined_load.py

```diff
@@ -25,18 +25,85 @@
 from gsy_e.gsy_e_core.exceptions import GSyException
 from gsy_e.gsy_e_core.global_objects_singleton import global_objects
 from gsy_e.gsy_e_core.util import should_read_profile_from_db
-from gsy_e.models.strategy.load_hours import LoadHoursStrategy
+from gsy_e.models.state import LoadState
+from gsy_e.models.strategy.load_hours import LoadHoursStrategy, LoadHoursEnergyParameters
+
+
+class DefinedLoadEnergyParameters(LoadHoursEnergyParameters):
+    """Energy parameters for the defined load strategy class."""
+    def __init__(self, daily_load_profile=None, daily_load_profile_uuid: str = None):
+        super().__init__(avg_power_W=0, hrs_per_day=24, hrs_of_day=list(range(0, 24)))
+        self.profile_uuid = daily_load_profile_uuid
+        self._load_profile_W = None
+        self._load_profile_kWh = {}
+        self.state = LoadState()
+
+        if should_read_profile_from_db(daily_load_profile_uuid):
+            self._load_profile_input = None
+        else:
+            self._load_profile_input = daily_load_profile
+
+    def serialize(self):
+        return {
+            "daily_load_profile": self._load_profile_input,
+            "daily_load_profile_uuid": self.profile_uuid
+        }
+
+    def event_activate_energy(self, area):
+        """
+        Runs on activate event.
+        :return: None
+        """
+        self.read_or_rotate_profiles()
+        super().event_activate_energy(area)
+
+    def reset(self, current_day: int, **kwargs) -> None:
+        if key_in_dict_and_not_none(kwargs, "daily_load_profile"):
+            self._load_profile_input = kwargs["daily_load_profile"]
+            self.read_or_rotate_profiles(reconfigure=True)
+
+    def read_or_rotate_profiles(self, reconfigure=False):
+        """Read power profiles or rotate them, from the DB or from JSON dicts."""
+        input_profile = (self._load_profile_input
+                         if reconfigure or not self._load_profile_W
+                         else self._load_profile_W)
```

Comment:
Nit, this could be a bit easier to understand
```suggestion
        input_profile = (self._load_profile_W
                         if reconfigure and _load_profile_W
                         else self. _load_profile_input)
```